setup-nat-routing.sh

```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+
+# Redirect stdout ( > ) into a log file.
+exec > /tmp/setup-nat-routing.log
+exec 2>&1
+
+set -e
+
+echo "NAT: Enabling routing features in the kernel"
+echo 1 | sudo tee /proc/sys/net/ipv4/ip_forward
+echo 0 | sudo tee /proc/sys/net/ipv4/conf/eth0/send_redirects
+sudo mkdir -p /etc/sysctl.d/
+echo 'net.ipv4.ip_forward = 1' | sudo tee /etc/sysctl.d/nat.conf
+echo 'net.ipv4.conf.eth0.send_redirects = 0' | sudo tee -a /etc/sysctl.d/nat.conf
+
+echo "NAT: Setting up NAT-MASQUERADE"
+sudo iptables -t nat -A POSTROUTING -o eth0 -s 0.0.0.0/0 -j MASQUERADE
+
+# Cloudinit changes the package repositories, which can make fail next steps.
+# Lets wait for it to finish.
+while pgrep cloud-init > /dev/null; do echo "Waiting for cloudinit to finish..."; sleep 1; done
+
+echo "Save iptables state"
+sudo apt-get update
+#sudo apt-get update | tee /tmp/second.txt # I need to do this twice!!! Why!?!?!?!
```

Comment:
What's this? Is it needed?

I think it'd be good to break the move to a single script and the modifications required to make it run on the new AWS AMI into separate commits, because it's difficult to tell what's changed here and where it's come from.